planetiler-custommap/samples/owg_simple.yml

```diff
@@ -0,0 +1,192 @@
+schemaName: OWG Simple Schema
+schemaDescription: Simple vector tile schema
+attribution: <a href="https://www.openstreetmap.org/copyright" target="_blank">&copy;
+  OpenStreetMap contributors</a>
+sources:
+- name: water_polygons
+  type: shapefile
+  url: https://osmdata.openstreetmap.de/download/water-polygons-split-3857.zip
+- name: osm
+  type: osm
+  area: geofabrik:rhode-island
+layers:
+- name: water
+  features:
+  - sources:
+    - osm
+    zoom:
+      minZoom: 0
+      maxZoom: 14
+    includeWhen:
+      geometry: polygon
+      tag:
+        key: natural
+        value:
+        - water
+    attributes:
+    - key: natural
+      constantValue: water
+      minZoom: 0
+    - key: intermittent
+      constantValue: 'true'
+      dataType: bool
+      includeWhen:
+        tag:
+          key: intermittent
+          value:
+          - true
+    - key: name
+      tagValue: name
+      dataType: string
+      includeWhen:
+        minTileCoverSize: 0.01
+      excludeWhen:
+        tag:
+          key: water
+          value:
+          - river
+          - canal
+          - stream
+  - sources:
+    - water_polygons
+    zoom:
+      minZoom: 0
+      minZoom: 14
+    includeWhen:
+      geometry: polygon
+    attributes:
+    - key: natural
+      constantValue: water
+      minZoom: 0
+  - sources:
+    - osm
+    zoom:
+      minZoom: 7
+      maxZoom: 14
+    includeWhen:
+      geometry: linestring
+      tag:
+        key: waterway
+        value:
+        - river
+        - stream
+        - canal
+    attributes:
+    - key: waterway
+      tagValue: waterway
+    - key: intermittent
+      constantValue: 'true'
+      dataType: bool
+      includeWhen:
+        tag:
+          key: intermittent
+          value:
+          - true
```

Comment:
Having removed the boolean data type stuff, it's now "yes" instead of true.  Do we want to infer booleans like that?

Reply:
I don't quite follow - I thought that `true`/`TRUE`/`True`/`yes` were all valid aliases for true in yaml?

Reply:
I was thinking that the most straightforward thing to do would be to treat all OSM tag value as strings and not try to do datatype conversions.  So e.g.:

```
      includeWhen:
        tag:
          key: intermittent
          value:
          - "yes"
```
The quotes would enforce treating the value as a string rather than boolean.

Reply:
Ahh got it, I see how we want the data type to coerce the string to a Boolean or number. If you use the getBoolean method on the feature it uses the same logic as imposm3 to determine if a string is truthy: https://github.com/onthegomap/planetiler/blob/2742bef48d09147d6b4e1f425ac7822882499887/planetiler-core/src/main/java/com/onthegomap/planetiler/util/Imposm3Parsers.java#L86

Reply:
Oh that makes things much easier.  It looks like we can support boolean, long, direction, and string in that way.

Reply:
Improved input data type implementation implemented in ceeb099

Reply:
Data typing resolved in ceeb099